api.proto

```diff
@@ -0,0 +1,43 @@
+syntax = "proto3";
+
+package api;
+
+service ResolverService {
+  rpc Apply(ApplyRequest)
+      returns (ApplyResponse);
+}
+
+message ApplyRequest {
+  SystemResolver resolver = 1;
+}
+
+message ApplyResponse {}
+
+// SystemResolver provides configuration options for the system resolver.
+//
+// https://man7.org/linux/man-pages/man5/resolv.conf.5.html.
+message SystemResolver{
```

Comment:
Given that DNS configuration may come from network configuration (DHCP), I wonder if maybe we should do DNS based on demand.

Reply:
We certainly need a way to mesh the two. In this case a user could submit static settings with this, and DHCP would be represented in this form on a GET. If that makes sense. So users can apply and get this, but DHCP is only GETable.